### PR TITLE
www-apache/mod_security: add 2.9.6

### PR DESCRIPTION
**Pull request by hydrapolic**
Tested on amd64.

```sh
--- mod_security-2.9.5-r1.ebuild        2022-07-13 07:06:05.813279772 +0000
+++ mod_security-2.9.6.ebuild   2022-11-18 17:54:54.775193893 +0000
@@ -36,7 +36,7 @@
 RDEPEND="${COMMON_DEPEND}
        geoip? ( dev-libs/geoip )
        mlogc? ( dev-lang/perl )"
-PDEPEND=">=www-apache/modsecurity-crs-2.2.6-r1"
+PDEPEND=">=www-apache/modsecurity-crs-3.3.2"
 
 S="${WORKDIR}/${MY_P}"
```

### CODE REVIEW

**Comment by hydrapolic** [comment]
Using in production.

**Review by juippis — COMMENTED** [review]
```
 * USE:        abi_x86_64 amd64 elibc_glibc geoip json kernel_linux lua lua_single_target_lua5-3 mlogc userland_GNU
..
 * QA Notice: The ebuild is installing to one or more unexpected paths:
 * 
 *   /mod_security2.so
 * 
 * Please fix the ebuild to use correct FHS/Gentoo policy paths.
```
O.O

can you try with these use flags?

**Comment by thesamesam** [comment]
That usually happens without the apache2 module eclass hack to use internals.

File: www-apache/mod_security/mod_security-2.9.6.ebuild [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+DESCRIPTION="Application firewall and intrusion detection for Apache"
+HOMEPAGE="https://github.com/SpiderLabs/ModSecurity"
+SRC_URI="https://github.com/SpiderLabs/ModSecurity/releases/download/v${PV}/${MY_P}.tar.gz"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+apache2 doc fuzzyhash geoip jit json lua mlogc"
+
+REQUIRED_USE="apache2
```
Comment (thesamesam) [inline www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Just make it always-on then?

**Comment by hydrapolic** [comment]
> ```
>  * USE:        abi_x86_64 amd64 elibc_glibc geoip json kernel_linux lua lua_single_target_lua5-3 mlogc userland_GNU
> ..
>  * QA Notice: The ebuild is installing to one or more unexpected paths:
>  * 
>  *   /mod_security2.so
>  * 
>  * Please fix the ebuild to use correct FHS/Gentoo policy paths.
> ```
> 
> O.O
> 
> can you try with these use flags?

Hello @juippis, the problem is that USE=apache2 wasn't set. I've pushed an update where now it's mandatory. Without USE=apache2 the paths are not set within the apache eclasses and it then installs into silly directories.

File: www-apache/mod_security/mod_security-2.9.6.ebuild (hunk 1 above)
Comment (hydrapolic) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Hello @thesamesam, how can that be expressed in the ebuild?

Comment (thesamesam) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Drop IUSE=apache2 & REQUIRED_USE, pass --enable-apache2 etc instead.

Alternatively, if you want to make it optional in future but it just doesn't work right now, we can add it in profiles/base/package.use.force. What do you prefer/intend?

Comment (hydrapolic) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: That's not passed to mod_security itself but used in https://github.com/gentoo/gentoo/blob/master/eclass/depend.apache.eclass#L193. 

I don't want to call the internals:
```
pkg_setup() {
  init_apache2
  init_apache2_late
```

But these are called from depend.apache only when apache2 is in IUSE (https://github.com/gentoo/gentoo/blob/master/eclass/depend.apache.eclass#L171).

There are 2 options:
1) https://gitweb.gentoo.org/repo/gentoo.git/commit/?id=ad49fd833d609fe1a93dd7b47df70a88dddd03d4
2) https://bugs.gentoo.org/616612#c13

Comment (thesamesam) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Ah, now I understand! Sorry, it seems obvious now you explained it.

It's fine if you use my method or if you call internals for now, as the eclass is broken anyway. I guess my preference is my method.

Comment (juippis) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: This is not a good solution, we should avoid using REQUIRED_USE as much as possible. As I see it, it's really no-op here, just an annoyance.

Comment (hydrapolic) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Just a side note, I started by the previous ebuild by using the internal calls (init_apache2 && init_apache2_late), but the CI report failed.

Comment (hydrapolic) [reply www-apache/mod_security/mod_security-2.9.6.ebuild:22]: Just checking https://gitweb.gentoo.org/repo/gentoo.git/tree/www-apache/mod_fastcgi_handler/mod_fastcgi_handler-0.6-r3.ebuild and it's also using the internal calls.

https://gitweb.gentoo.org/repo/gentoo.git/commit/www-apache/mod_fastcgi_handler?id=ad49fd833d609fe1a93dd7b47df70a88dddd03d4
https://gitweb.gentoo.org/repo/gentoo.git/commit/www-apache/mod_fastcgi_handler?id=9f513ee89c046aef6ca3c6f43ec999da0d2b4946

**Comment by hydrapolic** [comment]
Pushed an update with using the internal calls as before. Now the CI checks will fail with:

```sh
  IndirectInherits: version 2.9.6: depend.apache: indirect inherit usage: 'need_apache2', line 54
  InternalEclassUsage: version 2.9.6: depend.apache: internal usage: '_init_apache2', line 57
  InternalEclassUsage: version 2.9.6: depend.apache: internal usage: '_init_apache2_late', line 58
```

**Comment by thesamesam** [comment]
Yeah, that's fine, we just have to live with it for now. Thanks!

**Comment by hydrapolic** [comment]
Thanks!

**Pull request closed without merging**